cep-20.2.md

```diff
@@ -0,0 +1,832 @@
+The right-hand side of the key-value pair is a scalar (bool, number or string).
+
+The variables can reference other variables from the context section.
+
+> ![NOTE]
+> The order of the keys is not enforced by the YAML standard. We expect parsers to parse maps (especially the context section)
+> in the order they are defined in the file. However, we do not require this behavior for the recipe to be valid to conform to the YAML standard.
+> Given this, implementations need to ensure topological sorting is done before string interpolation.
+
+```yaml
+# The context dictionary defines arbitrary key-value pairs for Jinja interpolation
+# and replaces the {% set ... %} commands commonly used in recipes
+context:
+  variable: test
+  # note that we can reference previous values, that means that they are rendered in order
+  other_variable: test_${{ variable }}
+```
+
+
+
+## Package section
+
+```yaml
+package:
+  # The name of the package
+  name: string
+  # The version of the package, following the conda version spec
+  # Note that versions are always strings (1.23 is _not_ a valid version since it is a `float`. Needs to be quoted.)
+  version: string
+```
+
+## Build section
+
+```yaml
+build:
+  # the build number
+  number: integer  # defaults to 0
+
+  # the build string. This is usually ommited (can use `${{ hash }}` variable here)
+  string: string  # defaults to a build string made from "package hash & build number"
+
+  # A list of Jinja conditions under which to skip the build of this package (they are joined by `or`)
+  # What is valid in a an `if:` condition is valid
+  skip: [list of expressions]
+
+  # wether the package is a noarch package, and if yes, wether it is "generic" or "python"
+  noarch: OneOf<"generic" | "python">
+
+  # script can be a single string or a list of strings
+  # if script is a single string and ends with `.sh` or `.bat`, then we interpret it as a file
+  script: string | [string] | Script
+
+  # would love to get rid of this:
+  # merge the build and host environments (used in many R packages on Windows)
+  # was `merge_build_host`
+  merge_build_and_host_envs: bool (defaults to false)
+
+  # include files even if they are already in the environment
+  # as part of some other host dependency
+  always_include_files: [path]
+
+  # do not soft- or hard-link these files, but always copy them was `no_link`
+  always_copy_files: [glob]
+
+  variant:
+    # Keys to forcibly use for the variant computation (even if they are not in the dependencies)
+    use_keys: [string]
+
+    # Keys to forcibly ignore for the variant computation (even if they are in the dependencies)
+    ignore_keys: [string]
+
+    # used to prefer this variant less
+    # note: was `track_features`
+    down_prioritize_variant: integer (negative, defaults to 0)
+
+  # settings concerning only Python
+  python:
+    # List of strings, where each string follows this format:
+    # PythonEntryPoint: `bsdiff4 = bsdiff4.cli:main_bsdiff4`
```

Comment:
I never understood why this is not a map, but I guess the `:` in the syntax makes it confusing for the YAML parser (it would always need to be quoted?).